### PR TITLE
Big Refactoring; Code Cleanup; Renamed Singleton in Main; Removed sen…

### PR DESCRIPTION
**Pull request by TheRealJasmin**
…seless modifieres in interfaces; Some CodeStyle corrections; Added Validate.java for checking conditions like != null;

### CODE REVIEW

**Comment by WolfPlaysDE** [comment]
Nice work ;)

**Comment by TheRealJasmin** [comment]
Thank you ;)

File: src/de/janhektor/community/command/Argument.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 public abstract class Argument<T extends Plugin> {
 	
-	protected final T plugin;
```
Comment (Janhektor) [inline src/de/janhektor/community/command/Argument.java]: This line was used to access the plugin-instance from subclasses (every argument) simple without using the public getter. Maybe it's a good idea to let it <b>protected</b>.

Comment (TheRealJasmin) [inline src/de/janhektor/community/command/Argument.java]: But maybe it's not ;) Using the Getter would be a better style, but I wont prevent you from changing :)

File: src/de/janhektor/community/utils/Validate.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package de.janhektor.community.utils;
+
+import java.util.Objects;
+
+/**
+ * Erstellt von Jasmin am 05.07.2015 um 17:58
```
Comment (Janhektor) [inline src/de/janhektor/community/utils/Validate.java]: Please translate this note to English or remove it.
It's important to be consistent.

File: src/de/janhektor/community/command/Argument.java (hunk 1 above)
Comment (Janhektor) [inline src/de/janhektor/community/command/Argument.java]: OK.
We're going to work with the getter - if someone need's to access the plugin-instance multiply times in one class, it's possible to create a private member.

File: src/de/janhektor/community/utils/Validate.java (hunk 2 above)
Comment (TheRealJasmin) [inline src/de/janhektor/community/utils/Validate.java]: Translated and pushed.

File: src/de/janhektor/community/command/Argument.java (hunk 1 above)
Comment (TheRealJasmin) [inline src/de/janhektor/community/command/Argument.java]: Okay.

**Comment by Janhektor** [comment]
Hi Jasmin!

Thanks for your changes - good job.
It was really time to clean up some lines and make improvements (e.g. Validate class).
Everything is okay - I'll merge your fork into the official code-base (master branch).

Janhektor

**Comment by TheRealJasmin** [comment]
Thank you, I will continue working on this.

Regards
- Jasmin

**Comment by Aquaatic** [comment]
But... isn't there a Validate Class in Apache Commons?

**Comment by Janhektor** [comment]
Good thinking Aquaatic.
But we'll run into trouble if the package is changed.
In Spigot 1.8.3 there was a change such this: <b>org.bukkit.libs.com.google.gson</b> was moved to <b>com.google.gson</b>.
It's better to be independent.

**Comment by TheRealJasmin** [comment]
Yeah, there ate some Classes like that used in Bukkit&Spigot, but it's better to be independent, like Janhektor said and the classes by Apache Commons have more functions, than we need.

**Pull request merged**